### PR TITLE
Support embed.FS for HTML templates

### PR DESCRIPTION
**Pull request by yusufmalikul**
This PR will add support for `embed.FS` file system https://pkg.go.dev/embed. Useful for web apps that embed their template files.

### CODE REVIEW

File: html.go [hunk 1]
```diff
@@ -65,22 +75,14 @@ func (h *HTML) Load(names ...string) error {
...
-		if err != nil {
-			return err
-		}
 
-		_, err = clone.New("authboss").Funcs(h.funcMap).Parse(string(b))
+		FSPath := filepath.Join(h.overridePath, filename)
+		tpl, err := template.New(path.Base(FSPath)).Funcs(h.funcMap).ParseFS(h.templatesFS, FSPath)
 		if err != nil {
-			return errors.Wrapf(err, "failed to load template for page %s", n)
+			return errors.Wrapf(err, "failed to load embedded template for page %s", n)
```
Comment (wkoszek) [inline html.go]: @yusufmalikul Maybe let's keep the original string, because in the normal situation, this may not be `embedded` file?

Comment (yusufmalikul) [reply html.go]: that makes sense.

Comment (wkoszek) [reply html.go]: @yusufmalikul thanks

File: html.go [hunk 2]
```diff
@@ -65,22 +75,14 @@ func (h *HTML) Load(names ...string) error {
...
-			return err
-		}
-
-		clone, err := h.layout.Clone()
-		if err != nil {
-			return err
-		}
 
-		_, err = clone.New("authboss").Funcs(h.funcMap).Parse(string(b))
+		FSPath := filepath.Join(h.overridePath, filename)
```
Comment (wkoszek) [inline html.go]: @yusufmalikul Nitpicking: `FSPath` -> `fspath`.

**Comment by wkoszek** [comment]
@aarondl any chance you could merge it?

**Comment by aarondl** [comment]
I think I'm not understanding this PR. It seems to completely ignore the layout file? So it adds support for fs.FS but only for half of the library?

In this state this PR is not acceptable. A proper PR converting the lib to fs.FS support would:
* Fix or remove the `loadWithOverride` to support embed.FS
* Add `go:embed` lines to embed the existing pre-supplied templates
* Delete the bindata.go file
* Maintain backwards compatibility and all functionality (ie layouts should not break)

This is possible without being considered a breaking change so long as we keep the surface API backwards compatible because Go 1.16 is the lowest supported version of Go now.

I will close this as it's not currently mergeable.

**Pull request closed without merging**